apps/dashboard/app/javascript/dynamic_forms.js

```diff
@@ -198,8 +202,37 @@ function addHideHandler(optionId, option, key, configValue) {
   }
 
   updateVisibility({ target: document.querySelector(`#${optionId}`) }, changeId);
+};
+
+function newLabel(changeElement, key) {
+  const selectedOptionLabelIndex = changeElement[0].selectedIndex;
+  const selectedOptionLabel = changeElement[0].options[selectedOptionLabelIndex];
+  return selectedOptionLabel.dataset[key];
+};
```

Comment:
Does this only work for `select` options - I feel like we should support others like checkboxes. That's the only other type off the top I think this should support.

We can defer it, but it seems to me that someone may want `checkbox` support - and maybe other input types that I can't think of right now.